cranelift/codegen/src/legalizer/boundary.rs

```diff
@@ -1050,8 +1095,34 @@ fn spill_call_arguments(pos: &mut FuncCursor) -> bool {
     }
 
     // Insert the spill instructions and rewrite call arguments.
-    for (idx, arg, ss) in arglist {
-        let stack_val = pos.ins().spill(arg);
+    for (idx, arg, ss, size) in arglist {
+        let stack_val = if let Some(size) = size {
+            // Struct argument
+            let pointer_type = pos.func.dfg.value_type(arg);
+            let src = arg;
+            let dest = pos.ins().stack_addr(pointer_type, ss, 0);
+            let size = pos.ins().iconst(pointer_type, i64::from(size));
+            let signature = {
```

Comment:
Maybe pull out this and the `import_function` call below into an `import_memcpy` helper?

Also, it seems that this will create a new signature/import for every struct argument; can we lazily initialize it once instead?